### PR TITLE
5285 This test deletes and creates a new SMS GH source

### PR DESCRIPTION
**Pull request by pgarrison**
This test does the following
1. Make an SMS GH source
2. Generate 1 report from it [OK]
3. Delete the source
4. Make an SMS GH source
5. Generate 1 more SMS GH report [doesn't appear]

This test reuses a bit of code from `source-fetching-toggle.test.js`,
and these two should probably be worked together in some way.

### CODE REVIEW

**Comment by pgarrison** [comment]
Should this be 2.3 or 2.4?

File: test/end-to-end/smsgh.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  afterEach(utils.deleteSource.bind({}, 'SMS GH', 'foo'));
+  afterEach(utils.toggleFetching.bind({}, 'Off'));
+  afterEach(utils.resetBrowser);
+
+  var reqParams = {
+    from: '9845098450',
+    fulltext: 'loremipsumdolor',
+    date: '2016-09-01',
+    keyword: 'test'
+  };
```
Comment (pgarrison) [inline test/end-to-end/smsgh.test.js]: This shows up in the other smsgh test!

**Comment by pgarrison** [comment]
(This is also issue 5289)

**Comment by pgarrison** [comment]
Code climates complaints are wrong!

**Comment by karthiksbhat** [comment]
This issue was handled by 5102. This test is not breaking any more, and is a useful test to have.

**Pull request merged**